filters.go

```diff
@@ -95,6 +95,26 @@ func (p *Pipe) Exec(s string) *Pipe {
 	return q.WithReader(bytes.NewReader(output))
 }
 
+// ExecAt runs an external command at the specified directory and returns a pipe
+// containing the output. If the command had a non-zero exit status, the pipe's
+// error status will also be set to the string "exit status X", where X is the
+// integer exit status.
+func (p *Pipe) ExecAt(dir, s string) *Pipe {
+	if p == nil || p.Error() != nil {
+		return p
+	}
+	q := NewPipe()
+	args := strings.Fields(s)
+	cmd := exec.Command(args[0], args[1:]...)
+	cmd.Stdin = p.Reader
+	cmd.Dir = dir
```

Comment:
Can we have both Exec and ExecAt use the same basic machinery? It would be tiresome to have to fix a bug in multiple places, for example.